File: JAVA/CCPL/uj.java
// <3
// Tema: Math / Big Exponentiation
// Calcula N^D usando BigInteger para manejar resultados con exponentes muy grandes.

import java.util.*;
import java.math.*;

class Main {
    public static void main (String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        
        while (sc.hasNext()) {
            int N = sc.nextInt();
            int D = sc.nextInt();
            
            if (N == 0 && D == 0) break;
            
            System.out.println(BigInteger.valueOf(N).pow(D));
        }
    }
}

File: C++/CCPL/uj.cpp
// <3
// Tema: Math / Big Exponentiation
// Calcula p^q usando aritmética de precisión arbitraria (boost::multiprecision)
// para manejar resultados con exponentes muy grandes.

#include <bits/stdc++.h>
#include <boost/multiprecision/cpp_int.hpp>

using namespace std;
using namespace boost::multiprecision;

int main() {
    while (true) {
        long long p, q;
        cin >> p >> q;

        if (p == 0 && q == 0) return 0;

        cpp_int b = 1;

        for (long long i = 0; i < q; i++) {
            b *= p;
        }

        cout << b << "\n";
    }
}